### ISSUE TITLE
Creating a NDArray on GPU is very slow

### ISSUE THREAD

**Issue by EliasRas** [apache/incubator-mxnet#20531]
## Description
Program gets stuck when trying to call "mx.ndarray.ones" on a NVIDIA Turing GPU if mxnet version is >= 1.6.0. Program does not crash and GPU Memory usage keeps increasing.

Everything works if I use Pascal GPU (Quadro P400) or CPU. Everything works if I use mxnet 1.5.0.

### Error Message
No error message is given.

## To Reproduce
```
import os

with os.add_dll_directory(os.environ["CUDA_PATH"]):
    import mxnet as mx

testArray = mx.ndarray.ones((1,1),ctx=mx.gpu(1))
```

### Steps to reproduce

1. Install CUDA, Python
2. Install mxnet + dependencies using pip
3. Run the script using IDLE

## What have you tried to solve it?

1. Reinstalled CUDA, Python and mxnet
2. Tried various versions of CUDA and mxnet
3. Tried another version of cuDNN
4. Upgraded graphics drivers
5. Tried using the GPUs both in TCC mode and in WDDM mode

## Environment

- Windows Server 2019 Essentials
- Python 3.9.5
- pip 21.1.1

I've tried
- Quadro RTX 4000 and Tesla T4
- NVIDIA Graphics driver 441.66 and 471.11
- CUDA 9.2.148 and 10.2.86
- cuDNN v.7.6.5.32 and v8.2.2.26
- mxnet-cu92 1.6.0
- mxnet-cu102 1.6.0, 1.7.0 and 2.0.0b20201108

The mxnet packages are from [https://dist.mxnet.io/python](url)

**Comment by EliasRas** [comment]
Apparently I wasn't patient enough and the array is eventually created. However, this takes approx. 15 min and 1000 MiB of memory.

![HighMemoryUsage](https://user-images.githubusercontent.com/89061857/129848144-fcf7824a-3449-4919-94f5-d32e9dc01a56.png)

**Comment by TristonC** [comment]
@EliasRas, this is insane for being that slow. I am sorry to hear that. Is windows your only option to run MXNet?  @josephevans Does MXNet CI test windows platform?

**Comment by EliasRas** [comment]
> Is windows your only option to run MXNet?

@TristonC Unfortunately yes.

**Comment by EliasRas** [comment]
Is there a way for me to narrow-down the cause of this issue?

**Comment by ptrendx** [comment]
This sounds like a JIT compilation of PTX - @josephevans do you know which architectures we are targeting when building the Windows PIP wheel?

@EliasRas Does it happen during every launch of MXNet, or only during the first one?

**Comment by EliasRas** [comment]
Just the first time I call `mx.ndarray.ones`

**Comment by EliasRas** [comment]
@ptrendx Thank you for the hint. It took a while but I finally understood what you asked. It happened every time I launched MXNet but only due to inadequate cache size.

So yes, this was caused by compilation of PTX. My problem was solved by creating an environment variable "CUDA_CACHE_MAXSIZE". Default value 256 MiB was too low but 1 GiB seems to be enough, at least in my case. I also set "CUDA_CACHE_PATH" but that wasn't necessary.
